store.go

```diff
@@ -2117,20 +2151,43 @@ func (s *store) updateNames(id string, names []string, op updateNameOperation) e
 		return s.imageStore.updateNames(id, deduped, op)
 	}
 
-	// Check is id refers to a RO Store
+	// Check if the id refers to a read-only image store -- we want to allow images in
+	// read-only stores to have their names changed.
 	for _, is := range s.roImageStores {
 		store := is
 		if err := store.startReading(); err != nil {
 			return err
 		}
 		defer store.stopReading()
 		if i, err := store.Get(id); err == nil {
-			if len(deduped) > 1 {
-				// Do not want to create image name in R/W storage
-				deduped = deduped[1:]
+			// "pull up" the image so that we can change its names list
+			options := ImageOptions{
+				Metadata:     i.Metadata,
+				CreationDate: i.Created,
+				Digest:       i.Digest,
+				Digests:      copyDigestSlice(i.Digests),
 			}
```

Comment:
(Note to self: I didn’t check that this copies _everything_ relevant.)

Reply:
Hmm, `NamesHistory` probably needs to be copied, given that we "hide" the read-only version if we've got a writable image record that shares the same image ID, and information about the image apparently disappearing after a "pull up" was the problem I set out to fix.  Adding it.